BlueprintUI/Sources/Layout/Stack.swift

```diff
@@ -448,21 +454,90 @@ extension StackLayout {
-        let basisSizes = items.map { $0.content.measure(in: constraint).axis(on: axis) }
+        let minimumTotalSpacing = CGFloat(items.count - 1) * minimumSpacing
+
+        func basisSizes() -> [CGFloat] {
+            let constraint = vectorConstraint.constraint(axis: axis)
+
+            if BlueprintStackUpdates.isOptimizedLayoutEnabled {
+                final class Measurement {
+
+                    let item: (traits: Traits, content: Measurable)
+                    var size: CGSize?
+
+                    let isFixed: Bool
+
+                    var isFlexible: Bool {
+                        isFixed == false
+                    }
+
+                    init(item: (traits: StackLayout.Traits, content: Measurable)) {
+                        self.item = item
+                        size = nil
+
+                        isFixed = item.traits.growPriority == 0 && item.traits.shrinkPriority == 0
+                    }
+                }
+
+                let measurements: [Measurement] = items.map { item in
+                    Measurement(item: item)
+                }
+
+                // Measure fixed items
+
+                measurements.forEach { measurement in
+                    guard measurement.isFixed else { return }
 
-        func unconstrainedAxisSize() -> CGFloat {
-            let totalMeasuredAxis: CGFloat = basisSizes.reduce(0.0, +)
-            let minimumTotalSpacing = CGFloat(items.count - 1) * minimumSpacing
+                    measurement.size = measurement
+                        .item
+                        .content
+                        .measure(in: constraint)
+                }
+
+                // Determine how much space the fixed items took up
+
+                let fixedMagnitude: CGFloat = measurements.reduce(0) { magnitude, measurement in
+                    magnitude + (measurement.size?.axis(on: axis) ?? 0)
+                }
+
+                // From that, determine how much space the flexible items will have
+
+                let flexibleMagnitude: SizeConstraint.Axis = constraint.axis(on: axis) - fixedMagnitude - minimumTotalSpacing
+
+                let flexibleConstraint: SizeConstraint = {
+                    switch axis {
+                    case .horizontal:
+                        return .init(width: flexibleMagnitude, height: constraint.height)
+                    case .vertical:
+                        return .init(width: constraint.width, height: flexibleMagnitude)
+                    }
+                }()
+
+                // Measure the flexible items within that constraint
 
-            return totalMeasuredAxis + minimumTotalSpacing
+                measurements.forEach { measurement in
+                    guard measurement.isFlexible else { return }
+
+                    measurement.size = measurement
+                        .item
+                        .content
+                        .measure(in: flexibleConstraint)
```

Comment:
Ahh OK, I was wondering how you reduced measurement counts entirely within `_axisSegments`, but it seems to be the same number of calls. If I'm understanding right, it looks like this optimization really hinges on that `CacheTree` change in order to shortcut measurements from `_crossSegments` before they happen. The change to flexible child constraints makes the input values match for the 1-flexible-child case.

Reply:
Sorry, just saw this comment as I come back to this PR. The cache tree changes get us from 2 measurements to 1 – if you run the `test_measurementCounts` test and turn off `BlueprintStackUpdates.isOptimizedLayoutEnabled`, you can also see the main optimization, of some measurements happening 3, 4, times, etc.